pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -352,12 +352,18 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 		} else {
 			for _, zpool := range zpoolList {
 				rStorageInfo := new(info.StorageInfo)
+				rStorageInfo.ZpoolState = new(info.StorageState)
 				curentRaid := info.StorageRaidType_STORAGE_RAID_TYPE_NORAID
 				zpoolName := zpool.Properties[libzfs.PoolPropName].Value
+				storageState := zfs.GetZfsDeviceStatusFromStr(zpool.Properties[libzfs.PoolPropHealth].Value)
```

Comment:
Can there be odd cases when zpool.Properties[libzfs.PoolPropHealth] does not exist i.e., would it make sense to guard against that being nil?

Reply:
But conservative programming says one should check for such things to avoid a golang panic due to the key not being in the map.